frontend/workflows/projectSelector/src/state-store.tsx

```diff
@@ -0,0 +1,24 @@
+import type { ClutchError } from "@clutch-sh/core";
+import type { clutch as IClutch } from "@clutch-sh/api";
+import type { GroupState, State } from "./hello-world"
+import { Group } from "./hello-world"
+
+export default class StateStore{
+  [Group.PROJECTS]: GroupState;
+  [Group.UPSTREAM]: GroupState;
+  [Group.DOWNSTREAM]: GroupState;
+
+  projectData: { [projectName: string]: IClutch.core.project.v1.IProject };
+  loading: boolean;
+  error: ClutchError | undefined;
```

Comment:
looking into how we this class can extend the `State` interface we've already defined